app/src/main/java/ca/marshallasch/veil/MemoryStore.java

```diff
@@ -0,0 +1,521 @@
+
+            // make sure that it is a post item
+            if (wrapper.getType() == DhtProto.MessageType.POST) {
+               posts.add(wrapper.getPost());
+            }
+        }
+
+        // make sure that only 1 result was found
+        if (posts.size() > 1) {
+            throw new TooManyResultsException("Too many results for: " + hash);
+        } else if (posts.size() == 1) {
+            return new Pair<>(hash, posts.get(0));
+        }
+
+        // otherwise there were no results found.
+        return null;
+    }
+
+    /**
+     * This keyword must be a single token to search for, and it must match exactly.
+     * The keyword will be normalized to lowercase.
+     *
+     * @param keyword the keyword string to look up
+     * @return the list of results
+     */
+    @Override
+    @Nullable
+    public List<Pair<String, DhtProto.Post>> findPostsByKeyword(String keyword)
+    {
+        keyword = keyword.toLowerCase(Locale.getDefault());
+        ArrayList<DhtProto.DhtWrapper> entries = (ArrayList<DhtProto.DhtWrapper>)hashMap.get(Util.generateHash(keyword.getBytes()));
+
+        if (entries == null) {
+            return null;
+        }
+        ArrayList<String> postHashes = new ArrayList<>();
+
+        for (DhtProto.DhtWrapper wrapper: entries) {
+
+            // make sure that it is a post item
+            if (wrapper.getType() == DhtProto.MessageType.KEYWORD) {
+
+                // only include results that are for posts
+                DhtProto.KeywordType type = wrapper.getKeyword().getType();
+                if (type == TAG || type == TITLE_PARTIAL || type == TITLE_FULL) {
+                    postHashes.add(wrapper.getKeyword().getHash());
+                }
+            }
+        }
+
+        ArrayList<Pair<String, DhtProto.Post>> posts = new ArrayList<>();
+
+        // find all the post objects
+        for(String hash: postHashes) {
+
+            try {
+                posts.add(findPostByHash(hash));
+            }
+            catch (TooManyResultsException e) {
+                e.printStackTrace();
+            }
+        }
+
+        return posts;
+    }
+
+    /**
+     * This function will insert the comment object into the local hash table. This will also add
+     * record for looking up the comment by the post it is part of.
+     *
+     * @param comment the comment object that will be added
+     * @param postHash the post identifier that the comment is being added to
+     * @return the hash  identifying the comment
+     */
+    @Override
+    public String insertComment(DhtProto.Comment comment, String postHash)
+    {
+        String hash = Util.generateHash(comment.toByteArray());
+
+        DhtProto.DhtWrapper wrapper = DhtProto.DhtWrapper.newBuilder()
+                .setComment(comment)
+                .setType(DhtProto.MessageType.COMMENT)
+                .build();
+
+        insert(wrapper, hash);
+
+        // the keyword item is empty
```

Comment:
Make clear what this item is. Aka the search item for the comment you are about to insert.

Reply:
Fixed in commit e4bdcda